packages/search/src/util/merge-pagination/merge.ts

```diff
@@ -0,0 +1,41 @@
+import { InvalidPaginationInputError } from "./invalid-pagination-input";
+
+/**
+ * Interface that describes the name of a datasource (label) and the next start
+ * at which to search pages of data
+ */
+interface IDataPageNextStart {
+  label: string;
+  nextPageStart?: number;
+}
+
+/**
+ * When multiple data sources underly a paginated search, the search implementation should be able to
+ * search at the appropriate starting point for each source for a given paged set of results. This
+ * function merges an array of page states for each source, denoted by their label, into a base64-
+ * encoded cursor string that the implementation can use for the next set of results.
+ *
+ * @param pages an array of page states denoted by their label and the starting point for the next page
+ * @returns a base64-encoded cursor string to use when obtaining next page of results.
+ *
+ */
+export function mergePages(pages: IDataPageNextStart[]): string {
+  if (!pages || !Array.isArray(pages)) {
+    throw new InvalidPaginationInputError(
+      `Invalid Input Error. Must be array of IDataPageNextStart, received: ${JSON.stringify(
+        pages
+      )}`,
+      pages
+    );
+  }
+
+  const mergedPages: Record<string, number> = pages.reduce(
+    (pageObj: Record<string, number>, page: IDataPageNextStart) => {
+      pageObj[page.label] = page.nextPageStart || 1;
+      return pageObj;
+    },
+    {} as Record<string, number>
+  );
+
+  return Buffer.from(JSON.stringify(mergedPages)).toString("base64");
```

Comment:
AFAIK, `Buffer` is not a thing in a browser, so what's the plan there?

Reply:
@tomwayson - that also looked a little suss to me... but `test:ci` runs node + chrome and firefox tests, so I assumed it was cool... but yeah - looking in chome devtools... 
![image](https://user-images.githubusercontent.com/119129/103952510-c5c67e00-50fd-11eb-9bcb-4e0779e44a9c.png)
you are correct - `Buffer` is not a thing unless it's imported from something... 

which raises a few questions re: CI tests, in addition to the problem of how to handle the base64 encoding for this

Reply:
Since my PR's merged, I'm going to cut a release while this gets resolved

Reply:
@dbouwman if you haven't already please feel free to cut a release.

Looking at the node_modes there appears to be [buffer](https://github.com/feross/buffer) package being brought in. `npm list buffer` in the root shows it's a required dependency of `karma-typescript`.

In a [previous PR](https://github.com/Esri/hub.js/pull/442), I instead specified an external dependency to ensure compatibility with both browser and NodeJS. It was my understanding based on comments that I could refactor to use Buffer for node-specific functionality, but I can easily switch back to the relatively small library.

Reply:
I have updated the PR to use the external library for base64 encoding as shown in the previous PR.